app/db/interface.py

```diff
@@ -71,8 +75,30 @@ def update_payment(self, payment_id, payment_replacement=None, payment_attribute
         :param payment_replacement: <dict> a complete payload that describes a new payload which replaces the old one
         :param payment_attributes: <dict> a collection of new payment attribute values that will overwrite old ones
         """
-
-        raise NotImplementedError()
+        payment = self.db.session.query(Payment).get(payment_id)
+        if payment == None or payment.is_removed == True :
+            raise InvalidPaymentID('Invalid payment: Payment ID not found',status_code=404)
```

Comment:
better naming convention would have "Error" at the end of "InvalidPaymentID" just so its easy to see exactly what that is at a glance